index.rst

```diff
@@ -302,107 +305,152 @@ Is this case, we could have something like:
+    - These are not expected to be required as part of regular operations and are meant to be used when a non-standard configuration is required
+    - If an override configuration file is also site specific, then a prefix should be added indicating which site it belongs with (e.g. ``summit_reduced_stage_travel.yaml``)
 
 #.  The labels file (e.g. ``_labels.yaml``) shall continue to exist with the same format and purpose.
-#.  Additional configuration files can provide new values for individual configuration parameters.
 
-    - See this :download:`example configuration file <_static/ATSpectrograph_example_config.yaml>` for the ATSpectrograph CSC.
+        - No label shall exist having the name ``default``
+
+#.  If a CSC receives a ``start`` command with an empty ``configuration`` (see :ref:`renaming proposal <section-renaming>`) attribute, it shall load the values in ``_init.yaml`` then the site-specific file (e.g. ``_summit.yaml``).
+#.  If a CSC receives a ``start`` command with a ``configuration`` attribute equal to a label in ``_labels.yaml``, it shall load the values in ``_init.yaml``, then then the site-specific file (e.g. ``_summit.yaml``), and lastly the override file.
 
-#.  If a CSC receives a ``start`` command with an empty ``configuration`` (see :ref:`renaming proposal <section-renaming>`) attribute, it shall load the values in ``_base.yaml``.
-#.  If a CSC receives a ``start`` command with a ``configuration`` attribute equal to a label in ``_labels.yaml``, it shall load the values in ``_base.yaml`` first and override those values defined in the mapped configuration file.
-#.  The name ``default`` shall not be used for labels.
 #.  All valid configurations shall have a label to be loaded by the CSC.
-    Configurations cannot be loaded by filename.
-#.  The configuration repository should be kept clean of configurations used for unit testing or for different purposes (e.g. test stand configurations).
-    Configurations tailored for test stand can be kept in different branches/tags.
+
+#.  The configuration repository shall not contain configurations used for unit testing.
+    Override configurations that are site specific should contain the site name as a prefix to the filename (e.g. ``summit_simple_algorithm.yaml``).
     Configurations needed for unit testing shall be added to the ``test`` directory in the CSC repository and use the override feature in CSCs (see `Salobj documentation <https://ts-salobj.lsst.io>`__).
-#.  All configuration files shall have a header explaining that they are loading basic values from ``_base.yaml``, as shown in the :download:`example configuration file <_static/ATSpectrograph_example_config.yaml>` mentioned above.
+#.  All configuration files shall have a header metadata fields explaining that they are loading basic values from ``_init.yaml``, as shown in the :download:`example configuration file <_static/ATSpectrograph_example_config.yaml>` mentioned above.
```

Comment:
I strenuously object. This will add a huge amount of needless boilerplate.

In fact I much rather see that boilerplate removed from all the examples (and certainly the `_<site>.yaml` files).

In my opinion the description should focus on what is *important* about a given override file. That's what a give user needs to know. By adding all that common boilerplate I fear you will hide the information that really matters.

Reply:
The take-home point I'm trying to accomplish here is that the user needs to be reminded what this file is for and what it is doing.
We (unfortunately) can't just assume the user will have read the configuration workflow and know exactly how to handle/treat these files.  I'm open for other ideas on how we make that happen. Another option would be to have the first line of the file be a comment saying, "This is a configuration override file, for details on configuration management and usage see TSTN-017"  (or whatever document we should point them to).

Reply:
I am very uncomfortable with the idea that anyone would edit a config file who did not understand the workflow. How about a short README in each config package that gives the basics. I think if it's short and clear enough we can expect users to read that.

Then we can focus on explaining what is special about each override file in the metadata: why a user might choose this override file instead of the standard site-specific config.

Reply:
In fact that's my main point: the metadata for an override file should explain why a user might use the file. That's really the only thing that matters. The details of which file is read when should be left to an easy-to-find README or other basic docs that anyone messing with ts_config_x should be expected to know.